docs/api/portable-stories/portable-stories-vitest.mdx

```diff
@@ -161,27 +165,62 @@ draft: true
   ## setProjectAnnotations
 
   This API should be called once, before the tests run, typically in a [setup file](https://vitest.dev/config/#setupfiles). This will make sure that whenever `composeStories` or `composeStory` are called, the project annotations are taken into account as well.
+  
+  These are the configurations needed in the setup file:
+  - preview annotations: those defined in `.storybook/preview.ts`
+  - addon annotations (optional): those exported by addons
+  - beforeAll: code that runs before all tests
+  - testingLibraryRender: the render function from Testing Library
 
-  ```ts
-  // setup-portable-stories.ts
-  // Replace <your-renderer> with your renderer, e.g. react, vue3, svelte
-  import { setProjectAnnotations } from '@storybook/<your-renderer>';
-  import * as addonAnnotations from 'my-addon/preview';
-  import * as previewAnnotations from './.storybook/preview';
+  {/* prettier-ignore-start */}
 
-  setProjectAnnotations([previewAnnotations, addonAnnotations]);
-  ```
+  <CodeSnippets path="portable-stories-vitest-set-project-annotations.md" />
+
+  {/* prettier-ignore-end */}
 
   <Callout variant="warning">
-    Sometimes a story can require an addon's [decorator](../../writing-stories/decorators.mdx) or [loader](../../writing-stories/loaders.mdx) to render properly. For example, an addon can apply a decorator that wraps your story in the necessary router context. In this case, you must include that addon's `preview` export in the project annotations set. See `addonAnnotations` in the example above.
 
-    Note: If the addon doesn't automatically apply the decorator or loader itself, but instead exports them for you to apply manually in `.storybook/preview.js|ts` (e.g. using `withThemeFromJSXProvider` from [@storybook/addon-themes](https://github.com/storybookjs/storybook/blob/next/code/addons/themes/docs/api.md#withthemefromjsxprovider)), then you do not need to do anything else. They are already included in the `previewAnnotations` in the example above.
+  You **must** provide `testingLibraryRenderer` to use portable stories with Vitest.
+
   </Callout>
 
+  {/* TODO: Create issue for interest in non-Testing Library render option, with recipe, and mention here (Jest, too)
+    
+  ```js
+  import { render } from '@testing-library/[renderer]';
+  
+  setProjectAnnotations({
+    testingLibraryRender: render,
+  });
+  
+  For other testing renderers, you can configure renderToCanvas:
+  
+  import { render } from 'your-renderer';
+  
+  setProjectAnnotations({
+    renderToCanvas: ({ storyFn }) => {
+      const Story = storyFn();
+      
+      // Svelte
+      render(Story.Component, Story.props);
+      
+      // Vue
+      render(Story);
+      
+      // or for React
+      render(<Story/>);
+    },
+  });
+  ``` */}
```

Comment:
We should remove this and add a GH issue or reminder somewhere to do this part of the documentation
```suggestion
```